File: Spotter.Model/Responses/RecommendationResponse.cs
namespace Spotter.Model.Responses
{
    public class RecommendationResponse
    {
        public int EventId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string CategoryName { get; set; } = string.Empty;
        public string CategoryColorHex { get; set; } = string.Empty;
        public string? CoverImageUrl { get; set; }
        public DateTime StartsAt { get; set; }
        public string VenueName { get; set; } = string.Empty;
        public string CityName { get; set; } = string.Empty;
        public float Score { get; set; }
        public string Explanation { get; set; } = string.Empty;
    }
}

File: UI/spotter_mobile/lib/core/models/recommendation_response.dart
class RecommendationResponse {
  final int eventId;
  final String title;
  final String categoryName;
  final String categoryColorHex;
  final String? coverImageUrl;
  final DateTime startsAt;
  final String venueName;
  final String cityName;
  final double score;
  final String explanation;

  RecommendationResponse({
    required this.eventId,
    required this.title,
    required this.categoryName,
    required this.categoryColorHex,
    this.coverImageUrl,
    required this.startsAt,
    required this.venueName,
    required this.cityName,
    required this.score,
    required this.explanation,
  });

  factory RecommendationResponse.fromJson(Map<String, dynamic> json) {
    return RecommendationResponse(
      eventId: json['eventId'] as int,
      title: json['title'] as String,
      categoryName: json['categoryName'] as String,
      categoryColorHex: json['categoryColorHex'] as String,
      coverImageUrl: json['coverImageUrl'] as String?,
      startsAt: DateTime.parse(json['startsAt'] as String),
      venueName: json['venueName'] as String,
      cityName: json['cityName'] as String,
      score: (json['score'] as num).toDouble(),
      explanation: json['explanation'] as String,
    );
  }
}
